xdsl/transforms/DeadCodeElim.py

```diff
@@ -0,0 +1,21 @@
+from xdsl.traits import NoSideEffect
+from xdsl.pattern_rewriter import (PatternRewriter, PatternRewriteWalker,
+                                   RewritePattern, GreedyRewritePatternApplier)
+from xdsl.ir import MLContext, Operation
+from xdsl.dialects.builtin import ModuleOp
+
+
+class DCEImpl(RewritePattern):
+
+    def match_and_rewrite(self, op: Operation, rewriter: PatternRewriter, /):
```

Comment:
Potentially todo and issue to improve the annotation to handle `Union[Operation, NoEffect]` etc. I have a suspicion that if you write this then it'll actually already work out of the box

Reply:
You mean with the `op_type_rewrite_pattern` decorator? It doesn't since `if not issubclass(expected_type, Operation):` does not work then.

Reply:
I'm not sure I understand, what's the error when you run it? It does not process the code?